extensions/tika/deployment/src/main/java/io/quarkus/tika/deployment/TikaProcessor.java

```diff
@@ -0,0 +1,74 @@
+package io.quarkus.tika.deployment;
+
+import java.util.Arrays;
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import org.apache.tika.detect.Detector;
+import org.apache.tika.detect.EncodingDetector;
+import org.apache.tika.parser.Parser;
+
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.substrate.RuntimeInitializedClassBuildItem;
+import io.quarkus.deployment.builditem.substrate.ServiceProviderBuildItem;
+import io.quarkus.deployment.builditem.substrate.SubstrateResourceBuildItem;
+import io.quarkus.deployment.util.ServiceUtil;
+
+public class TikaProcessor {
+    private static final Set<String> NOT_NATIVE_READY_PARSERS = Arrays.stream(new String[] {
+            "org.apache.tika.parser.mat.MatParser",
+            "org.apache.tika.parser.journal.GrobidRESTParser",
+            "org.apache.tika.parser.journal.JournalParser",
+            "org.apache.tika.parser.jdbc.SQLite3Parser",
+            "org.apache.tika.parser.mail.RFC822Parser",
+            "org.apache.tika.parser.pkg.CompressorParser",
+            "org.apache.tika.parser.geo.topic.GeoParser"
+    }).collect(Collectors.toSet());
+
+    @BuildStep
+    public void produceRuntimeInitializedClasses(BuildProducer<RuntimeInitializedClassBuildItem> resource) {
+        //org.apache.tika.parser.pdf.PDFParser (https://issues.apache.org/jira/browse/PDFBOX-4548)
+        resource.produce(new RuntimeInitializedClassBuildItem("org.apache.pdfbox.pdmodel.font.PDType1Font"));
+    }
+
+    @BuildStep
+    public void produceTikaCoreResources(BuildProducer<SubstrateResourceBuildItem> resource) throws Exception {
+        resource.produce(new SubstrateResourceBuildItem("org/apache/tika/mime/tika-mimetypes.xml"));
+        resource.produce(new SubstrateResourceBuildItem("org/apache/tika/parser/external/tika-external-parsers.xml"));
+    }
+
+    @BuildStep
+    public void produceTikaParsersResources(BuildProducer<SubstrateResourceBuildItem> resource) throws Exception {
+        resource.produce(new SubstrateResourceBuildItem("org/apache/tika/parser/pdf/PDFParser.properties"));
+    }
+
+    @BuildStep
+    public void producePdfBoxResources(BuildProducer<SubstrateResourceBuildItem> resource) throws Exception {
```

Comment:
Not sure if it's required for Tika extension, but when I was trying to make PDFBox work in quarkus, when I was calling:
```
        PDFTextStripper stripper = new PDFTextStripper();
```
I was receiving this error inside native image:

https://gist.github.com/agiertli/35170cfdd5a8b3fd7e86fc5d6e238e3a

I solved it by adding Identity-h to the native-image resources:

https://github.com/cvt-oss/fb-invoice-pdf-analyzer/blob/quarkus/pom.xml#L112
Dur

Reply:
@agiertli thanks for the review and the hint. I don't recall right now but without those (or one of those) inclusions I was getting some native error as well.
It depends on whether the `PDFTextStripper` is used on the Tika path somewhere or not. We may need to take care of the `Identity-HS` resources as well.

Can you consider giving this PR a try and see if it works for your case ? We may get some valuable early review this way, thanks

Reply:
It;s better to have the extension add these resource files instead of forcing the user to set them in the pom.ml for info. That's one the reason they exist, hide the GraalVM ugliness.

Reply:
@emmanuelbernard I agree :-) but @agiertli  has not used Tika but PDFBox directly so I'm not sure that is needed for the Tika code path yet

Reply:
**Update**: `Identity-HS` is [here](https://github.com/apache/pdfbox/tree/trunk/fontbox/src/main/resources/org/apache/fontbox/cmap). It is hard right now to get all the resources which will actually be needed so I propose to add them whenever the exception is raised  at the Tika path.

Reply:
I confirm that parsing the same PDF using tika-extension API succeed:
```
      TikaParser parser = new TikaParser();
      String invoiceText = parser.getContent(is).getText();
```

This extension also simplifies the project configuration a lot! **No more hacks** like this are needed anymore:
https://github.com/gunnarmorling/quarkus-pdf-extract/blob/master/pom.xml#L113

The only difference in behavior is that  the above code produces string with extra \n lines (while using original PDFBox API does not do this)

Reply:
Hi @agiertli thanks, `Identity-HS` and friends may still show up on some Tika code paths for some complex PDF files :-), but we will take care of those resources if needed for sure

Reply:
For info in Hobernate ORM we had a long list of files ot add and we used the following script that we run one a new version of Hibernate is integrated and adapt the processor accordingly
https://github.com/quarkusio/quarkus/blob/master/extensions/hibernate-orm/deployment/src/test/java/io/quarkus/hibernate/orm/HqlNodeScannerTestCase.java

Might be an inspiration

Reply:
Hi Emmanuel, yes, something like that would be needed to smartly add all the resources that may be needed (as far as I understand the idea, based on the link, would be to iterate over all, in this case, non-class resources and add them)

Reply:
In practice, we generated that list and copied it in our extension code. So that we don't have to iterate over this list every single build. But that's a burden when the extension updates to the next version of the underlying library to make sure to rerun that script. It happens relatively rarely though